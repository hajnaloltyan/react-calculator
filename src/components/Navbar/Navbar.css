.navbar {
  width: 20vw;
}

.navlinks {
  display: flex;
  align-items: center;
  justify-content: space-between;
  list-style: none;
}

a {
  font-size: 20px;
  text-decoration: underline transparent;
  color: #000;
  font-weight: 600;
}

ul {
  padding: 0;
}

.active {
  color: #d600c4;
  text-decoration-color: #d600c4;
  text-underline-offset: 10px;
  text-decoration-thickness: 3px;
}
