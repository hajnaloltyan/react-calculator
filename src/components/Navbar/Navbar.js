import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';

const links = [
  { path: '/', text: 'Home' },
  { path: '/calculator', text: 'Calculator' },
  { path: '/quotes', text: 'Quotes' },
];

const Navbar = () => (
  <nav className="navbar">
    <ul>
      {links.map(({ path, text }) => (
        <li key={text}>
          <NavLink
            to={path}
          >
            {text}
          </NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Navbar;
