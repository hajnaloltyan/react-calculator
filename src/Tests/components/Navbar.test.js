import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../../components/Navbar/Navbar';

describe('Test the Navbar component', () => {
  test('Check if the component render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    ).toJSON;
    expect(tree).toMatchSnapshot();
  });

  test('Check how many links are and they content', () => {
    const Navbarcomponent = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const links = Navbarcomponent.container.querySelectorAll('a');

    expect(links.length).toBe(3);
    expect(links[0]).toHaveTextContent('Home');
    expect(links[1]).toHaveTextContent('Calculator');
    expect(links[2]).toHaveTextContent('Quotes');
  });
});
