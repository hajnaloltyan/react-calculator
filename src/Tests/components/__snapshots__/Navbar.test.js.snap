// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Navbar component Check if the component render correctly 1`] = `[Function]`;
